packages/core/scripts/local/ClaimAllRewards.js

```diff
@@ -1,72 +1,190 @@
+//   older ones. Defaults to 1.2.2 and latest.
 
-const Voting = artifacts.require("Voting");
-const VotingInterfaceTesting = artifacts.require("VotingInterfaceTesting");
+const { getAbi, getAddress } = require("../../index");
 const TransactionBatcher = artifacts.require("TransactionBatcher");
+const lodash = require("lodash");
+const winston = require("winston");
+const { GasEstimator } = require("@uma/financial-templates-lib");
 
-const argv = require("minimist")(process.argv.slice(), { string: ["round", "batcherAddress"] });
+const argv = require("minimist")(process.argv.slice(), {
+  string: ["round", "batcherAddress", "claimAddress", "excludeAddress", "version"],
+  boolean: "onlyPrint"
+});
 
-const { toBN, toWei } = web3.utils;
+const { toBN, toWei, toChecksumAddress } = web3.utils;
 
 // This script claims all voter's rewards for the round provided.
 async function claimRewards() {
-  const voting = await VotingInterfaceTesting.at((await Voting.deployed()).address);
+  const rounds = argv.round ? lodash.castArray(argv.round) : [];
+  const excludeAddresses = argv.excludeAddress ? lodash.castArray(argv.excludeAddress) : [];
+  const claimAddresses = argv.claimAddress ? lodash.castArray(argv.claimAddress) : [];
+  const versions = argv.version ? lodash.castArray(argv.version) : ["latest", "1.2.2"];
 
-  const events = await voting.contract.getPastEvents("VoteRevealed", {
-    filter: { roundId: argv.round },
-    fromBlock: 0,
-    toBlock: "latest"
+  const account = (await web3.eth.getAccounts())[0];
+  const networkId = await web3.eth.net.getId();
+
+  const votingContracts = versions.map(version => {
+    if (version.startsWith("1")) {
+      return new web3.eth.Contract(getAbi("Voting", version), getAddress("Voting", networkId, version));
+    } else {
+      return new web3.eth.Contract(
+        getAbi("VotingAncillaryInterfaceTesting", "latest"),
+        getAddress("Voting", networkId, version)
+      );
+    }
   });
 
-  const votersToPriceRequests = {};
+  const events = lodash.flatten(
+    await Promise.all(
+      votingContracts.map(voting =>
+        voting.getPastEvents("VoteRevealed", {
+          filter: { roundId: rounds, voter: claimAddresses },
+          fromBlock: 0,
+          toBlock: "latest"
+        })
+      )
+    )
+  );
+
+  const priceRequestMap = {};
   for (const event of events) {
     const voter = event.returnValues.voter;
-    const newPriceRequest = { identifier: event.returnValues.identifier, time: event.returnValues.time };
-    if (votersToPriceRequests[voter]) {
-      votersToPriceRequests[voter].push(newPriceRequest);
+    const round = event.returnValues.roundId;
+    const contractAddress = event.address;
+    const key = [voter, round, contractAddress].join("|");
+    if (excludeAddresses.includes(toChecksumAddress(voter))) {
+      console.log("Found noclaim voter", voter);
+      continue;
+    }
+    const newPriceRequest = {
```

Comment:
was trying to understand why price requests have anything to do with this, but i think i realized. is it the case that even governance votes are also considered price requests? so this basically covers all votes

Reply:
Yep, exactly!